### ISSUE TITLE
Infinite recursion not detected in lambdas

### ISSUE THREAD

**Issue by cscalfani** [elm-lang/elm-compiler#1577]
The following code compiles without an error:

```elm
b =
    (\_ -> b) ()
```

Generating the following code:

```js
var b = function (_p20) {
	return b;
}({ctor: '_Tuple0'});
```

Turns out that b will be undefined. See example code: https://ellie-app.com/Jc7tR3gpHka1/0

**Comment by evancz** [comment]
Read https://github.com/elm-lang/elm-compiler/issues/873 to learn about the kinds of things that can be detected. It is known that other cases cannot be detected. Another example would be:

```elm
loop x = loop x
```

Figuring out what kind of recursion is "good" is an undecidable problem in general. If you have a specific proposal of a practical *mechanism* to detect more "bad" cases, please write it up and share it on elm-dev.
